src/transactions/PathPaymentOpFrame.cpp

```diff
@@ -286,4 +286,39 @@ PathPaymentOpFrame::doCheckValid(uint32_t ledgerVersion)
     }
     return true;
 }
+
+std::unordered_set<LedgerKey>
+PathPaymentOpFrame::getLedgerKeysToPrefetch()
+{
+    std::unordered_set<LedgerKey> keys;
+    keys.emplace(stellar::accountKey(mPathPayment.destination));
+
+    std::vector<Asset> allAssets{mPathPayment.sendAsset,
```

Comment:
this is a bit wasteful to allocate a vector of `Asset`: you can probably just refactor this to use a function that processes an Asset and performs whatever is in the loop, and call the function on `sendAsset`, `destAsset` and using `std::for_each` on the path itself